runtime/pkg/jsonval/jsonval.go

```diff
@@ -0,0 +1,382 @@
+		// (They don't have a native JSON representation)
+		if math.IsNaN(float64(v)) || math.IsInf(float64(v), 0) {
+			return nil, nil
+		}
+		return float64(v), nil
+	case float64:
+		// Turning NaNs and Infs into nulls until frontend can deal with them as strings
+		// (They don't have a native JSON representation)
+		if math.IsNaN(v) || math.IsInf(v, 0) {
+			return nil, nil
+		}
+		return v, nil
+	case time.Time:
+		if t != nil && t.Code == runtimev1.Type_CODE_DATE {
+			return v.In(time.UTC).Format(time.DateOnly), nil
+		}
+		return v, nil
+	case string:
+		if !utf8.ValidString(v) {
+			return nil, fmt.Errorf("invalid UTF-8 in string: %q", v)
+		}
+		return v, nil
+	case []byte:
+		if t != nil && t.Code == runtimev1.Type_CODE_UUID {
+			uid, err := uuid.FromBytes(v)
+			if err == nil {
+				return uid.String(), nil
+			}
+		}
+		// Falling through for default handling
+	case big.Int:
+		// Evil cast to float until frontend can deal with bigs:
+		v2, _ := new(big.Float).SetInt(&v).Float64()
+		return v2, nil
+		// This is what we should do when frontend supports it:
+		// return v.String(), nil
+	case *big.Int:
+		// Evil cast to float until frontend can deal with bigs:
+		v2, _ := new(big.Float).SetInt(v).Float64()
+		return v2, nil
+		// This is what we should do when frontend supports it:
+		// return v.String(), nil
+	case duckdb.Decimal:
+		// Evil cast to float until frontend can deal with bigs:
+		denom := big.NewInt(10)
+		denom.Exp(denom, big.NewInt(int64(v.Scale)), nil)
+		v2, _ := new(big.Rat).SetFrac(v.Value, denom).Float64()
+		return v2, nil
+	case map[string]any:
+		var t2 *runtimev1.StructType
+		if t != nil {
+			t2 = t.StructType
+		}
+		return toMap(v, t2)
+	case []any:
+		return toSlice(v, t)
+	case map[any]any:
+		var t2 *runtimev1.MapType
+		if t != nil {
+			t2 = t.MapType
+		}
+		return toMapCoerceKeys(v, t2)
+	case duckdb.Map:
+		return ToValue(map[any]any(v), t)
+	case duckdb.Interval:
+		return map[string]any{"months": v.Months, "days": v.Days, "micros": v.Micros}, nil
+	case net.IP:
+		return v.String(), nil
+	case *net.IP:
+		if v != nil {
+			return ToValue(*v, t)
+		}
+		return nil, nil
+	case *bool:
+		if v != nil {
+			return *v, nil
+		}
+		return nil, nil
+	case *int:
+		if v != nil {
+			return int64(*v), nil
+		}
+		return nil, nil
+	case *int8:
+		if v != nil {
+			return int64(*v), nil
+		}
+		return nil, nil
+	case *int16:
+		if v != nil {
+			return int64(*v), nil
+		}
+		return nil, nil
+	case *int32:
+		if v != nil {
+			return int64(*v), nil
+		}
+		return nil, nil
+	case *int64:
+		if v != nil {
+			return *v, nil
+		}
+		return nil, nil
+	case *uint:
+		if v != nil {
+			return uint64(*v), nil
+		}
+		return nil, nil
+	case *uint8:
+		if v != nil {
+			return uint64(*v), nil
+		}
+		return nil, nil
+	case *uint16:
+		if v != nil {
+			return uint64(*v), nil
+		}
+		return nil, nil
+	case *uint32:
+		if v != nil {
+			return uint64(*v), nil
+		}
+		return nil, nil
+	case *uint64:
+		if v != nil {
```

Comment:
I think we should serialize it as string instead of uint64 to avoid loss of precision. Some information here : https://groups.google.com/g/protobuf/c/4-BY-k-Lk-g

Reply:
Yeah that would be nice, but the frontend can't handle numbers as strings yet. That's also why we have this for all the numeric types:
```go
// Evil cast to float until frontend can deal with bigs:
```

Reply:
Does UI parse the CSV/JSON exports ?
I thought this is only for CSV/JSON exports.

Reply:
Regarding frontend – this will also be integrated for the resolvers and custom APIs, so will eventually also be exposed there.